File: src/test/java/br/com/jiratorio/factory/AccountFactory.java
package br.com.jiratorio.factory;

import br.com.jiratorio.domain.vo.Account;
import br.com.jiratorio.domain.vo.CurrentUser;
import br.com.jiratorio.service.TokenService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class AccountFactory {

    public static final String DEFAULT_USER = "default_user";

    @Autowired
    private TokenService tokenService;

    public Account defaultUser() {
        return buildUser(DEFAULT_USER);
    }

    public Account buildUser(final String username) {
        return new Account(username, "secret-token",
                new CurrentUser("name", "email@company.com"));
    }

    public String defaultUserToken() {
        return tokenService.encode(defaultUser());
    }
}

File: src/test/java/br/com/jiratorio/factory/entity/AccountFactory.kt
package br.com.jiratorio.factory.entity

import br.com.jiratorio.domain.Account
import br.com.jiratorio.domain.CurrentUser
import br.com.jiratorio.service.TokenService
import org.springframework.stereotype.Component

@Component
class AccountFactory(
        private val tokenService: TokenService
) {

    fun defaultUser() =
            buildUser(defaultUserName())

    fun buildUser(username: String) =
            Account(username, "secret-token", CurrentUser("name", "email@company.com"))

    fun defaultUserToken() =
            tokenService.encode(defaultUser())

    fun defaultUserName() =
            "default_user"

}
